src/main/java/uk/gov/cabinetoffice/csl/controller/reset/ResetFormValidator.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.cabinetoffice.csl.controller.reset;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+@Component
+public class ResetFormValidator implements Validator {
+
+    private final String passwordPattern;
+
+    @Autowired
+    public ResetFormValidator(@Value("${accountValidation.passwordPattern}") String passwordPattern) {
+        this.passwordPattern = passwordPattern;
+    }
+
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return ResetForm.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
```

Comment:
Maybe add a few tests for the `validate()` method in the test classes just to make sure they throw the correct errors where appropriate

Reply:
Following unit tests are added in the "ResetControllerTest" to validate the password:
shouldShowPasswordInvalidErrorMessageIfNumberIsMissingFromPassword
shouldShowPasswordInvalidErrorMessageIfLowerCaseCharacterMissingFromPassword
shouldShowPasswordInvalidErrorMessageIfUpperCaseCharacterMissingFromPassword
shouldShowPasswordInvalidErrorMessageIfPasswordLengthIsLessThan8Characters
shouldShowPasswordInvalidErrorMessageIfPasswordHasSpace
shouldShowPasswordMismatchErrorMessageIfPasswordAndConfirmPasswordMismatch